pkg/spoke/addon/configuration.go

```diff
@@ -0,0 +1,222 @@
+package addon
+
+import (
+	"crypto/sha256"
+	"crypto/x509/pkix"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	operatorhelpers "github.com/openshift/library-go/pkg/operator/v1helpers"
+	certificatesv1beta1 "k8s.io/api/certificates/v1beta1"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+const (
+	installationNamespaceAnnotation = "installNamespace"
+	bootstrapSecretAnnotation       = "bootstrapSecret"
+	registrationAnnotation          = "registrations"
+	clusterNamePlaceHolder          = "{ClusterName}"
+	agentNamePlaceHolder            = "{AgentName}"
+)
+
+// registrationConfig contains necessary information for addon registration. A addon may have multiple
+// registrationConfigs.
+type registrationConfig struct {
```

Comment:
You may want this to be a public class that shared with leases part. FYI @skeeey

Reply:
public is unnecessary, we should be in the same package

Reply:
could we consider this to
```
type registration struct {
    SignerName            string      `json:"signerName,omitempty"`
Subject               certSubject `json:"subject,omitempty"`

}

type registrationConfig struct {
AddonName             string      `json:"addonName"`
	InstallationNamespace string      `json:"installNamespace"`
	BootstrapSecretName   string      `json:"bootstrapSecretName,omitempty"`
        Registrations []registration
}
```

thus `getRegistrationConfigs` will return one registrationConfig for one addon